### ISSUE TITLE
Add label RING_ID to corosync metric

### ISSUE THREAD

**Issue by MalloZup** [ClusterLabs/ha_cluster_exporter#16]
Create a metric
`corosync_ring_errors("ring_id=1") 1
`corosync_ring_errors("ring_id=42") 0

 here so we can point what ring is failing.  A zero or 1 is a failure.

**Comment by MalloZup** [comment]
@diegoakechi  to me we should consider : 

this is an output

```
Printing ring status.
	Local node ID 16777226
	RING ID 0
			id      = 10.0.0.1
			status  = Marking ringid 0 interface 10.0.0.1 FAULTY
	RING ID 1
			id      = 172.16.0.1
			status  = ring 1 active with no faults																				  			```

Now if we want to have  a label by ring id, which ID should be take?
RING ID 0 or `	id      = 10.0.0.1`

Also right now we have the total of failure. To me this could be as minimalist approach . 

If we need more precision we can do it

**Comment by gao-yan** [comment]
Indeed the output of corosync-cfgtool is kind of confusing. How about adding a new field "ring_address" or combining the information like "0 (address: 10.0.0.1)"?

**Comment by MalloZup** [comment]
@gao-yan to me we could add it yes. I just hope the output stay the same since we will use it as api😅

**Comment by MalloZup** [comment]
@gao-yan  can a ring have multiple adress? :thinking:

**Comment by MalloZup** [comment]
adding this would require some anti-pattern things in prometheus. I am not sure if we need it ..

**Comment by gao-yan** [comment]
I don't think output of corosync-cfgtool changes often. But the output between corosync-2.4 and corosync-3 are different:

2.4:
https://github.com/corosync/corosync/blob/needle-2.4/tools/corosync-cfgtool.c#L68

3.0:
https://github.com/corosync/corosync/blob/master/tools/corosync-cfgtool.c#L90

But we will likely take corosync-3 only starting from SLE 16.

In corosync-2.4, corosync-cmapctl outputs the info in better format, for example some relevant information from there:

runtime.totem.pg.mrp.rrp.0.faulty (u8) = 1
runtime.totem.pg.mrp.srp.members.1084783184.ip (str) = r(0) ip(192.168.122.80) r(1) ip(127.0.0.1)
totem.interface.0.bindnetaddr (str) = 192.168.122.0

I don't have corosync-3 in hands, but IIRC, rrp (Redundant Ring Protocol) is even dropped. https://jira.suse.com/browse/PM-1203

@yuanren10 should have better understanding/suggestions on the topics in here :-)

**Comment by gao-yan** [comment]
> @gao-yan can a ring have multiple adress?
I don't think so. A ring/interface can only be configured with one "bindnetaddr".

**Comment by MalloZup** [comment]
ok thx @gao-yan . So I think for a first corosync 0.5 version of the exporter, I would go with only the metric total. which is the most simple one

For the other metric we need to research a bit if we needed it or not to rely.

`corosync-cmapctl` seems a valid tool to me more handy to parse. If it is stable we might just use that.

**Comment by gao-yan** [comment]
It makes sense to provide only the most basic and important info for the first step.

And let's check how things and output fields from corosync-cmapctl are different between corosync-2.4 and corosync-3 so that we have better ideas on the options.

**Comment by MalloZup** [comment]
yes agree. thx @gao-yan :rocket:

**Comment by ReyRen** [comment]
Same with mentioned by @gao-yan , in corosync2.4.+, link status seems only can be tell by return code of "corosync-cfgtool" and "runtime.totem.pg.mrp.rrp.0.faulty".

**Comment by MalloZup** [comment]
@ReyRen thx so far. I think for moment this issue is not super urgent but thx for all info was helpfull :rocket:

**Comment by mbothorel** [comment]
Hi, I can confirm the issue using corosync 3+ in Debian10.
Any plan to support it ? How can I help ?

**Comment by MalloZup** [comment]
@mbothorel Hi. 

If you want to help on this, you need to create a new metric with some labels.
Some doc:

https://github.com/ClusterLabs/ha_cluster_exporter/blob/master/doc/design.md

Also a part of setting up the development env., check my first comment on the issue which give an hint what  we need to achieve.

If you have any question feel free to ping me. There is no stupid question, :grin: 

If you wanna work on it, I can assign this to you.

Let me know and thank you for proposing it!
